client/ayon_core/lib/attribute_definitions.py

```diff
@@ -215,10 +216,11 @@ def deserialize(cls, data):
 
 
 # -----------------------------------------
-# UI attribute definitoins won't hold value
+# UI attribute definitions won't hold value
 # -----------------------------------------
 
 class UIDef(AbstractAttrDef):
+    type = "ui-def"
```

Comment:
This is UI abstract class, should not implement what should implement children classes.
```suggestion
```